### ISSUE TITLE
Grandparent rollup not triggering when intermediate (parent) object is reparented

### ISSUE THREAD

**Issue by baobao917** [jamessimone/apex-rollup#518]
Opening up this new issue that was identified/discussed in #509 after it was fixed/closed.

In #509 I had the following question:
![image](https://github.com/jamessimone/apex-rollup/assets/104730171/c8f9faf0-9afa-4ee9-b35f-026f57750256)

Grandparent rollup relationship:
ProductConsumed->Work Order->Asset.Plan_Spend__c
- Work Order-Product Consumed = Master-Detail relationship
- Work Order-Asset = Lookup relationship

As stated in the screenshot, I tested this two different ways:
- Without a flow trigger on intermediate object (work order)
- With a flow trigger on intermediate object (work order)

In neither case, did it trigger the grandparent rollup to calculate.  
Please see video - https://www.loom.com/share/cb5ddf8df4694b94a16632988da40995?sid=49bb8813-8e49-4200-b0c1-8c143ceca6bb

**Comment by jamessimone** [comment]
@baobao917 made some progress on this issue tonight, though I still have some work to do on it. This will require the intermediate flow on Work Item - which I've activated in your sandbox. At present, the recalculation for the new grandparent is working; what I have left to do is the proper handling for the old grandparent record.

**Comment by baobao917** [comment]
@jamessimone Thanks....quick question on the flow trigger on the intermediate object (work order).  That particular object has lots of automations on it and we've been really careful about about overloading it b/c we've been hitting limits in the past.  This is one reason we felt ApexRollup was well suited for this use case (grandparent rollup) which would bypass the intermediate object altogether.

Since we need a flow/apex trigger on the intermediate object to handle possible intermediate reparenting, I'm assuming by adding a flow trigger on the work order object that could contribute to that transaction time and possibly hit our limits again.  Do you have any suggestions for us? Would making the intermediate trigger an apex trigger instead of a flow trigger significantly make a difference in performance?

**Comment by jamessimone** [comment]
Yeah, for Work Order in particular I totally understand that concern. Apex will always be slightly more performant than Flow, and it's possible that could end up making the difference on an automation-heavy object like that, but I was looking at a few logs generated and the Flow wasn't really adding much of anything in the way of processing time there.

**Comment by jamessimone** [comment]
This should be fixed in v1.6.1

**Comment by baobao917** [comment]
@jamessimone Thanks! I tested and when reparenting the intermediate object to a new grandparent, the new grandparent rollup calculation was triggered.

I only noticed one follow-up and wasn't sure how you wanted me to handle or if it's part of issue #514 .  The old grandparent didn't recalculate.  Here's a video.

https://www.loom.com/share/7018d6cf5c6c4a209410e755ebf57e0c?sid=27fe8391-21ab-4dad-b4e8-1a0babf2a426
